packages/main/src/components/ThemeProvider/index.tsx

```diff
@@ -16,47 +16,61 @@ export interface ThemeProviderProps {
 
 const generateClassName = createGenerateClassName();
 
-export class ThemeProvider extends PureComponent<ThemeProviderProps> {
-  static defaultProps = {
-    withToastContainer: false
-  };
-
-  constructor(props) {
-    super(props);
-    boot().then((_) => {
-      let existingThemingProperties = document.querySelector('head style[data-ui5-webcomponents-theme-properties]');
-      if (!existingThemingProperties || !existingThemingProperties.innerHTML) {
-        injectThemeProperties(fiori3ThemeProperties._);
+const ThemeProvider: FC<ThemeProviderProps> = (props) => {
+  const theme = getTheme();
+  useEffect(() => {
+    boot().then(async () => {
+      // TODO will rename to 'data-ui5-theme-properties' after next UI5 Web Components Release
+      const styleElement = document.head.querySelector('style[data-ui5-webcomponents-theme-properties]');
+      // only inject parameters for sap_fiori_3 and if they haven't been injected before
+      if (theme === Themes.sap_fiori_3 && !styleElement.textContent) {
+        injectThemeProperties(fiori3Theme);
+        const CSSVarsPonyfill = window['CSSVarsPonyfill'];
+        if (Device.browser.msie && CSSVarsPonyfill) {
+          setTimeout(() => {
+            CSSVarsPonyfill.resetCssVars();
+            CSSVarsPonyfill.cssVars();
+          }, 0);
+        }
       }
     });
-  }
+  }, [theme]);
+  const { withToastContainer, children } = props;
 
-  private static getTheme = (theme: Themes) => {
+  const isCompactSize = getCompactSize();
+
+  const contentDensity = useMemo(() => {
```

Comment:
useMemo is not really needed here as contentDensity is a string and === hits on equal strings.